File: webpage/src/shared/api/assessment.ts
import { fetchWithAuth } from "./client";

export interface Assessment {
  id: number;
  member_id: number;
  status: "draft" | "in_progress" | "completed";
  goal: string | null;
  risk_level: "low" | "medium" | "high" | null;
  questionnaire_summary: Record<string, any> | null;
  report_summary: Record<string, any> | null;
  created_at: string;
  updated_at: string;
  completed_at: string | null;
}

export interface AssessmentCreatePayload {
  goal?: string;
  questionnaire_summary?: Record<string, any>;
}

export interface AssessmentCompletePayload {
  risk_level: "low" | "medium" | "high";
  report_summary: Record<string, any>;
}

export interface AssessmentReport {
  id: number;
  status: string;
  risk_level: string | null;
  report_summary: Record<string, any> | null;
  completed_at: string | null;
}

export async function createAssessment(payload: AssessmentCreatePayload): Promise<Assessment> {
  const response = await fetchWithAuth("/assessments", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  const result = await response.json();
  return result.data as Assessment;
}

export async function getAssessments(status?: string): Promise<Assessment[]> {
  const url = status ? `/assessments?status=${encodeURIComponent(status)}` : "/assessments";
  const response = await fetchWithAuth(url);
  const result = await response.json();
  return result.data as Assessment[];
}

export async function getAssessment(id: number): Promise<Assessment> {
  const response = await fetchWithAuth(`/assessments/${id}`);
  const result = await response.json();
  return result.data as Assessment;
}

export async function completeAssessment(id: number, payload: AssessmentCompletePayload): Promise<Assessment> {
  const response = await fetchWithAuth(`/assessments/${id}/complete`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  const result = await response.json();
  return result.data as Assessment;
}

export async function getAssessmentReport(id: number): Promise<AssessmentReport> {
  const response = await fetchWithAuth(`/assessments/${id}/report`);
  const result = await response.json();
  return result.data as AssessmentReport;
}

File: Rogers/app/schemas/assessment.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class AssessmentBase(BaseModel):
    goal: str | None = Field(default=None, description="评估目标")
    questionnaire_summary: dict[str, Any] | None = Field(default=None, description="问卷摘要")


class AssessmentCreate(AssessmentBase):
    pass


class AssessmentUpdate(BaseModel):
    status: str | None = None
    risk_level: str | None = None
    questionnaire_summary: dict[str, Any] | None = None
    report_summary: dict[str, Any] | None = None
    goal: str | None = None


class AssessmentComplete(BaseModel):
    risk_level: str = Field(..., description="风险等级(low, medium, high)")
    report_summary: dict[str, Any] = Field(..., description="评估报告内容")


class AssessmentResponse(AssessmentBase):
    id: int
    member_id: int
    status: str
    risk_level: str | None = None
    report_summary: dict[str, Any] | None = None
    created_at: datetime
    updated_at: datetime
    completed_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class AssessmentReportResponse(BaseModel):
    id: int
    status: str
    risk_level: str | None
    report_summary: dict[str, Any] | None
    completed_at: datetime | None

    model_config = ConfigDict(from_attributes=True)
